zebra/zserv.c

```diff
@@ -521,9 +553,8 @@ static void zserv_process_messages(struct event *thread)
 
 	stream_fifo_free(cache);
 
-	/* Reschedule ourselves if necessary */
-	if (need_resched)
-		zserv_event(client, ZSERV_PROCESS_MESSAGES);
+	if (client_ibuf_fifo_cnt == p2p)
```

Comment:
continuing with the comments from above, I'd rather see us allow more reads when possible, even if the fifo isn't empty. It's cleaner to pace the reads, rather than get into a stop-and-wait situation. If zebra's not busy, we can be responsive as possible, and if zebra is busy, we will be working hard in 'process' anyway.

Reply:
1) Earlier without this change, we iterate the zserv_process_messages() only packet_to_process(configurable/default-1000) times and reschedule ourselves. This is because ibuf_fifo can have >p2p/1000 elements.

2) With this change, ibuf_fifo can NEVER have more than packet_to_process/def-1000 items. So at any given time, we can only process max items of ibuf_fifo.  Once processed, unless zserv_read adds more items in the fifo(with this change it signals main), there is no point in rescheduling. 

In case Zebra is fast, it still works the same as before except now we are not hoarding more memory in zebra by growing ibuf_fifo items and expect the client IO to pulse main when it adds more items on the fifo.